lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -121,7 +136,18 @@ namespace Js
             }
         }
 
-        JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedRegExp, propertyName);
+        JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedRegExp, varName);
+    }
+
+
+    RecyclableObject* JavascriptRegExp::GetThisObject(Arguments& args, PCWSTR varName, ScriptContext* scriptContext)
+    {
+        if (args.Info.Count == 0 || !JavascriptOperators::IsObject(args[0]))
```

Comment:
Looks like it would be at least 1, but it isn't only here we compare against `0`. I'll keep the check for now, but I've created #387 to track this.